### DISCUSSION TITLE
Bosch C1200 Low Quality Receiving

### DISCUSSION THREAD

**Discussion by Raybom** [emsesp/EMS-ESP32#1469] [Q&A] [answered]
**Problem Description:**
I have custom hardware build. Connected to EMS Bus but Rx reading are very low quality and very slow. Also there is no Tx readings.

**Expected Result:** 
%100 Quality

**Current Result:**
%12 Quality

**ESP32:**
ESP32-DevKitM-1
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/hw-reference/esp32/user-guide-devkitm-1.html

[emsesp_info.txt](https://github.com/emsesp/EMS-ESP32/files/13488030/emsesp_info.txt) 

[log.txt](https://github.com/emsesp/EMS-ESP32/files/13488033/log.txt)

![image](https://github.com/emsesp/EMS-ESP32/assets/27249574/45fbeb36-16fd-44a8-8f20-df1814aa19ad)

![image](https://github.com/emsesp/EMS-ESP32/assets/27249574/74604f31-4bc2-423e-a244-69ea29efc095)

![image](https://github.com/emsesp/EMS-ESP32/assets/27249574/7bcc66f8-5888-4822-a8ca-893c2e76c905)

**Comment by MichaelDvP** [comment]
Please add some information about the custom hardware.

**Reply by Raybom** [reply]
![Schematic_EMS-ESP (1)](https://github.com/emsesp/EMS-ESP32/assets/27249574/11f4e11b-2509-4dab-8965-73950171fad5)

This one with ESP32. Also i checked signal input and output. Compared and it is working fine.

**Comment by bbqkees** [comment, marked as answer]
Do not power anything from the bus. The EMS bus is too weak to power an ESP32 board. This will cause RX to drop.
Just remove the voltage regulator and connect a USB power supply to the ESP board.

**Reply by Raybom** [reply]
Thanks for the reply. I connect ESP32 from external supply and it worked.
